### PR TITLE
Added a way to append new scopes to existing default scopes

### PR DESCRIPTION
**Pull request by saturnism**
For #248

### CODE REVIEW

File: spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/test/java/org/springframework/cloud/gcp/core/autoconfig/GcpContextAutoConfigurationTests.java [hunk 1]
```diff
@@ -55,13 +59,46 @@ public void testGetProjectIdProvider_withGcpProperties() {
 		assertEquals("test-project", provider.getProjectId());
 	}
 
+	@Test
+	public void testStarterScopes() {
+	}
```
Comment (joaoandremartins) [inline spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/test/java/org/springframework/cloud/gcp/core/autoconfig/GcpContextAutoConfigurationTests.java]: Did you mean to add anything here?

File: spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/test/java/org/springframework/cloud/gcp/core/autoconfig/GcpContextAutoConfigurationTests.java [hunk 2]
```diff
@@ -55,13 +59,46 @@ public void testGetProjectIdProvider_withGcpProperties() {
...
+		assertEquals(scopes.size(), 1);
+		assertTrue(scopes.contains("myscope"));
+	}
+
+	@Test
+	public void testResolveScopesStarterScopesPlaceholder() {
+		loadEnvironment("spring.cloud.gcp.credentials.scopes=STARTER_SCOPES,myscope");
+		GcpContextAutoConfiguration configuration = this.context.getBean(GcpContextAutoConfiguration.class);
+		List<String> scopes = configuration.resolveScopes();
+		assertTrue(scopes.size() > 1);
```
Comment (joaoandremartins) [inline spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/test/java/org/springframework/cloud/gcp/core/autoconfig/GcpContextAutoConfigurationTests.java]: We could test the exact number of scopes. Testing if there is more than one scope isn't extremely useful.

File: spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/autoconfig/GcpContextAutoConfiguration.java [hunk 3]
```diff
@@ -56,26 +60,48 @@
...
+					resolvedScopes.addAll(CREDENTIALS_SCOPES_LIST);
+				}
+				else {
+					resolvedScopes.add(scope);
+				}
+			});
+
+			return Collections.unmodifiableList(new ArrayList<>(resolvedScopes));
+		}
+		else {
```
Comment (joaoandremartins) [inline spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/autoconfig/GcpContextAutoConfiguration.java]: Nit: could probably get rid of this else

File: spring-cloud-gcp-docs/src/main/asciidoc/core.adoc [hunk 4]
```diff
@@ -114,3 +114,12 @@ property.
 ----
 spring.cloud.gcp.credentials.scopes=https://www.googleapis.com/auth/pubsub,https://www.googleapis.com/auth/sqlservice.admin
 ----
+
+You can also use `STARTER_SCOPES` placeholder as a scope to represent the starters default scopes,
+and append the additional scopes you need to add.
+
+[source,yaml]
+----
+spring.cloud.gcp.credentials.scopes=STARTER_SCOPES,https://www.googleapis.com/auth/cloud-vision
```
Comment (artembilan) [inline spring-cloud-gcp-docs/src/main/asciidoc/core.adoc]: Why `STARTER`? Typically it is like `STANDARD`...

**Comment by joaoandremartins** [comment]
@artembilan @saturnism went on vacations for a couple of months, so I think we would better merge this PR as is and I can open a new one to fix the issues with this one.

WDYT?

**Comment by artembilan** [comment]
Fine with me.

Thanks

**Comment by meltsufin** [comment]
+1 for post-merge fix

File: spring-cloud-gcp-docs/src/main/asciidoc/core.adoc (hunk 4 above)
Comment (joaoandremartins) [reply spring-cloud-gcp-docs/src/main/asciidoc/core.adoc]: I think I even prefer `DEFAULT_SCOPES` here...

**Pull request merged**